Intersect.Server/Entities/Player.cs

```diff
@@ -812,7 +812,13 @@ public override void Die(bool dropItems = true, Entity killer = null)
             {
                 base.Die(dropItems, killer);
             }
-            
+
+            if (Options.Player.XPLossOnDeath.Equals(true))
+            {
+                var ExpLoss = (GetExperienceToNextLevel(this.Level) * (Options.Player.XPLossPercent / 100.0));
```

Comment:
This isn't calculating based on the current xp...

instead you want:
var ExpLoss = (this.Exp * (Options.Player.XPLossPercent / 100.0));

Reply:
But the XP loss is calculated based on the total required to level and not the xp that the player currently has, that was purposeful, I believe that in most games the XP loss calculation is the percentage over what he needs to upar and not what he already has, no?

Reply:
I might be wrong I but I don't think that's how everyone will want it.... how about one more boolean in the options?

public bool ExpLossFromCurrentExp {get; set;} = true

Then you can do an if/then to calculate exp loss based on both?

Reply:
Great idea I will change the files here, and let the staff choose the mode

Reply:
I think its all ok now 👯